### PR TITLE
Document multiple public libraries

### LINKED ISSUE #6362: Visibility field not listed under cabal spec changes.
The cabal docs list the [`visibility`](https://cabal.readthedocs.io/en/latest/developing-packages.html#pkg-field-library-visibility) field as existing as of Cabal 3.0, however the list of format changes doesn't list it as an addition in [3.0](https://cabal.readthedocs.io/en/latest/file-format-changelog.html#cabal-version-3-0)

### PR DESCRIPTION
**Pull request by fgaz**
About time I write this :D

Closes #6362

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary. :heavy_check_mark: :heavy_check_mark: :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: doc/cabal-package.rst [hunk 1]
```diff
@@ -991,6 +997,18 @@ a real-world use case:
 
       default-language: Haskell2010
 
+**Multiple public libraries**
+
+Cabal 3.0 and later support exposing multiple libraries from a single package
+through the field :pkg-field:`visibility`.
+Having multiple public libraries is useful for separating the unit of
+distribition (package) from the unit of buildable code (library).
```
Comment (jneira) [inline doc/cabal-package.rst:1005]: minor typo
```suggestion
distribution (package) from the unit of buildable code (library).
```

File: doc/cabal-package.rst [hunk 2]
```diff
@@ -1561,16 +1579,16 @@ system-dependent values for these fields.
...
-    main library component; in future, packages with multiple exposed
-    public library components will be supported and a syntax for
-    referring to public sub-libraries will be provided).
-
-    In order to specify an intra-package dependency on an internal
-    library component you can use the unqualified name of the
-    library component. Note that locally defined sub-library
-    names shadow external package names of the same name. See section on
-    :ref:`Internal Libraries <sublibs>` for examples and more information.
+    provided by, optionally followed by a colon and the library name
```
Comment (jneira) [inline doc/cabal-package.rst:1582]: Is this correct?

Comment (fgaz) [reply doc/cabal-package.rst:1582]: as far as I can see yes, `pkg:lib`

File: doc/cabal-package.rst (hunk 1 above)
Comment (fgaz) [reply doc/cabal-package.rst]: Thanks, fixed

**Review by jneira — APPROVED** [review]
perfect, thanks for improving docs

**Comment by jneira** [comment]
I suppose it is fine to backport it to 3.8

**Review by Mikolaj — APPROVED** [review]

**Comment by Mikolaj** [comment]
> If you don't mind I'm going to skip the 2 day waiting period, so that 3.8 will have multilibs docs in time for the discourse announcement

That's a good idea.

**Comment by fgaz** [comment]
@Mergifyio rebase

**Comment by Kleidukos** [comment]
@fgaz ah, thank you so much for this PR, I just stumbled across the passive-aggressive sentences that you fix and I was about to raise a ticket :D

**Comment by fgaz** [comment]
@Kleidukos you're welcome, but what was passive-aggressive about that?

**Comment by Kleidukos** [comment]
> @Kleidukos you're welcome, but what was passive-aggressive about that?

The ellipses at the end of each sentence :p

**Pull request merged**